pkg/tarfs/tarfs.go

```diff
@@ -715,32 +693,52 @@ func (t *Manager) DetachLayer(snapshotID string) error {
 	return nil
 }
 
-func (t *Manager) getSnapshotStatus(snapshotID string, lock bool) (*snapshotStatus, error) {
+func (t *Manager) getSnapshotStatus(snapshotID string) (*snapshotStatus, error) {
 	t.mutex.Lock()
 	defer t.mutex.Unlock()
 	st, ok := t.snapshotMap[snapshotID]
 	if ok {
-		if lock {
-			st.mutex.Lock()
-		}
+		st.mutex.Lock()
```

Comment:
Why lock this by default, Can we add lock and release it after calling `getSnapshotStatus`?

Otherwise, change `getSnapshotStatus` to `getSnapshotStatusWithLock`?

Reply:
It's ensure atomicity by acquiring the lock on `snapshotStatus` with the `Manager.mutex" locked.
So renamed `getSnapshotStatus` to `getSnapshotStatusWithLock`.